crates/nargo_cli/src/cli/execute_cmd.rs

```diff
@@ -57,29 +61,87 @@ fn execute_with_path<B: Backend>(
     prover_name: String,
     compile_options: &CompileOptions,
 ) -> Result<(Option<InputValue>, WitnessMap), CliError<B>> {
-    let CompiledProgram { abi, circuit } = compile_circuit(backend, program_dir, compile_options)?;
+    let (compiled_program, context) = compile_circuit(backend, program_dir, compile_options)?;
+    let CompiledProgram { abi, circuit, debug } = compiled_program;
 
     // Parse the initial witness values from Prover.toml
     let (inputs_map, _) =
         read_inputs_from_file(program_dir, prover_name.as_str(), Format::Toml, &abi)?;
 
-    let solved_witness = execute_program(backend, circuit, &abi, &inputs_map)?;
-
+    let solved_witness = execute_program(backend, circuit, &abi, &inputs_map, &debug, &context)?;
     let public_abi = abi.public_abi();
     let (_, return_value) = public_abi.decode(&solved_witness)?;
 
     Ok((return_value, solved_witness))
 }
 
+fn extract_unsatisfied_constraint_from_nargo_error(nargo_err: &NargoError) -> Option<usize> {
+    let solving_err = match nargo_err {
+        nargo::NargoError::SolvingError(err) => err,
+        _ => return None,
+    };
+
+    match solving_err {
+        acvm::pwg::OpcodeResolutionError::UnsatisfiedConstrain { opcode_label } => {
+            match opcode_label {
+                OpcodeLabel::Unresolved => {
+                    unreachable!("Cannot resolve index for unsatisifed constraint")
+                }
+                OpcodeLabel::Resolved(opcode_index) => Some(*opcode_index as usize),
+            }
+        }
+        _ => None,
+    }
+}
+fn report_unsatisfied_constraint_error(
+    opcode_idx: Option<usize>,
+    debug: &DebugInfo,
+    context: &Context,
+) {
+    if let Some(opcode_index) = opcode_idx {
+        if let Some(loc) = debug.opcode_location(opcode_index) {
+            let errs_warnings: ErrorsAndWarnings = vec![FileDiagnostic {
+                file_id: loc.file,
+                diagnostic: CustomDiagnostic::simple_error(
+                    "Unsatisfied constraint".to_string(),
+                    "happening on this line".to_string(),
+                    loc.span,
+                ),
+            }];
+            noirc_errors::reporter::report_all(&context.file_manager, &errs_warnings, false);
+        }
+    }
+}
+
 pub(crate) fn execute_program<B: Backend>(
     backend: &B,
     circuit: Circuit,
     abi: &Abi,
     inputs_map: &InputMap,
+    debug: &DebugInfo,
+    context: &Context,
 ) -> Result<WitnessMap, CliError<B>> {
     let initial_witness = abi.encode(inputs_map, None)?;
 
-    let solved_witness = nargo::ops::execute_circuit(backend, circuit, initial_witness)?;
+    let solved_witness_err = nargo::ops::execute_circuit(backend, circuit, initial_witness);
```

Comment:
We should make use of `execute_program_without_debug` here instead of duplicating the function

Reply:
Done